modules/bibupload/lib/batchuploader_engine.py

```diff
@@ -62,6 +62,66 @@
 PERMITTED_MODES = ['-i', '-r', '-c', '-a', '-ir',
                    '--insert', '--replace', '--correct', '--append', '--holdingpen']
 
+FILE_SEARCH_PATTERN = [
+    # Example: 1234567.pdf
+        ('recid',
+        re.compile(r'(?i)(^\d+)\.pdfa?$'),
+        ('recid', )),
+    ## Example: Pages from C88-01-23_15-24.pdf
+        #('cnum and page range with prefix',
+        #re.compile(r'^Pages from (C\d\d-\d\d-\d\d)[-_](\d+\-\d+)\.pdfa?$'),
+        #('773__w', '773__c')),
+    ## Example: Pages from C88-01-23_15-24.pdf
+    ##PLEASE REMOVE WHITESPACE FROM FILENAMES FIRST! OR implement that in this script
+        ('cnum and page range with prefix but take only start page',
+        re.compile(r'^Pages_from_(C\d\d-\d\d-\d\d)[-_](\d+\-)\d+\.pdfa?$'),
+        ('773__w', '773__c')),
+    # Example: Pages from C75-03-02_101.pdf
+        ('cnum and page start with prefix with optional dot',
+        re.compile(r'^Pages_from_(C\d\d-\d\d-\d\d?.?\d)[-_](\d+)\.pdfa?$'),
+        ('773__w', '773__c')),
+    ## Example: 'Pages from C77-03-06.1_93.pdf'
+        #('cnum and page start with prefix with dot',
+        #re.compile(r'^Pages_from_(C\d\d-\d\d-\d\d\.\d)[-_](\d+)\.pdfa?$'),
+        #('773__w', '773__c')),
+    # Example: Pages from C88-03-06.1_79-89.pdf
+        ('cnum and page range with prefix but take only start page with optional dot',
+        re.compile(r'^Pages_from_(C\d\d-\d\d-\d\d?.?\d)[-_](\d+\-)\d+\.pdfa?$'),
+        ('773__w', '773__c')),
+    # Example: C73-03-04_Proceedings.pdf
+        ('cnum with Proceedings suffixed with optional dot',
+        re.compile(r'^(C\d\d-\d\d-\d\d?.?\d)[-_](Proceedings).pdfa?$'),
+        ('773__w', '980__a')),
+    # Example: c12-03-17.1-p578.pdf
+        ('cnum and page',
+        re.compile(r'(?i)(^C\d\d-\d\d-\d\d.?\d*)[-_]p0*(\d+)\.pdfa?$'),
+        ('773__w', '773__c')),
+    # Example: c12-03-17.1-p578_comment.pdf
+        ('cnum and page',
+        re.compile(r'(?i)(^C\d\d-\d\d-\d\d.?\d*)[-_]p0*(\d+)_(.*)\.pdfa?$'),
+        ('773__w', '773__c')),
+    # Example: c12-03-17.2-p578-582.pdf
+        ('cnum and page-range',
+        re.compile(r'(?i)(^C\d\d-\d\d-\d\d.?\d*)[-_]p0*(\d+)-0*\d+\.pdfa?$'),
+        ('773__w', '773__c')),
+    # Example: 1234567-p578.pdf
+        ('recid and page',
+        re.compile(r'(?i)(^\d+)[-_]p0*(\d+)\.pdfa?$'),
+        ('773__0', '773__c')),
+    # Example: 1234567-p578-591.pdf
+        ('recid and page-range',
+        re.compile(r'(?i)(^\d+)[-_]p0*(\d+)-0*d+\.pdfa?$'),
+        ('773__0', '773__c')),
+    # Example: RepNo_CMS-pas123.pdf
+        ('report-number',
+        re.compile(r'(?i)^RepNo[_-](.*)\.pdfa?$'),
+        ('reportnumber', )),
+    # Example: anything.pdf
```

Comment:
L119-122:
the garbage collection works only within a known context.
I.e. you need to know which record is collecting this stuff.
Are you sure this makes sense in general?
I guess you have to make an exception and raise a warning.